components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -350,3 +354,39 @@ func assertIssuerIsReachable(host string) error {
 	}
 	return nil
 }
+
+func assertIssuerProvidesDiscovery(issuer string) error {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		Proxy:           http.ProxyFromEnvironment,
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   2 * time.Second,
+		// never follow redirects
+		CheckRedirect: func(*http.Request, []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	resp, err := client.Get(issuer + "/.well-known/openid-configuration")
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("OIDC Discovery is not provided.")
```

Comment:
Would it make sense to provide more details in the response, such that users have an easier time debugging why this failed?

Reply:
Changed to express the expectation.